src/components/database.rs

```diff
@@ -146,26 +147,35 @@ impl DatabaseComponentImplementation for DatabaseComponent {
                 "postgres://{}:{}@{}/{}",
                 self.db_user, self.db_password, self.db_host, self.db_name
             );
-            log::debug!("DB URL: {}", url);
+            log::info!("DB URL: {}", url);
 
             let pool = PgPoolOptions::new().min_connections(5).max_connections(10);
 
-            let db_connection = match pool.connect(url.as_str()).await {
+            let log_level = std::env::var("RUST_LOG")
+                .map(|level| LevelFilter::from_str(&level).unwrap_or(LevelFilter::Info))
+                .unwrap_or(LevelFilter::Info);
+
+            let options = PgConnectOptions::from_str(url.as_str())
+                .expect("Unable to parse Database URL")
+                .log_statements(log_level)
+                .clone();
+
+            let db_connection = match pool.connect_with(options).await {
                 Ok(db) => db,
                 Err(err) => {
                     log::debug!("Error on connecting to DB: {:?}", err);
```

Comment:
Shouldn't this one be `error`?
```suggestion
                    log::error!("Error on connecting to DB: {:?}", err);
```